lib/screens/MainScreen.dart

```diff
@@ -112,8 +111,12 @@ class _MainScreenState extends State<MainScreen> {
   }
 
   Widget _buildSites() {
+    if (sites == null || sites!.length == 0) {
+      return _buildNoSites();
+    }
```

Comment:
just noting that I pulled this in here and removed `_buildNoSites`. I felt like it was safer to have the null-check _inside_ the function that accesses `sites`.